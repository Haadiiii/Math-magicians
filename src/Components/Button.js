const Button = (props) => (
  <div className="button">
    <button type="submit" onClick={props.clickHandler}>
      {props.name}
    </button>
  </div>
);

export default Button;
