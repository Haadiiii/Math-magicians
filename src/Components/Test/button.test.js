import React from "react";
import Button from "../Button";
import { render, cleanup } from "@testing-library/react";
import renderer from "react-test-renderer";

afterEach(cleanup);

it("render Without crashing", () => {
  render(<Button name="me" />);
});

it("matches snapshot for button", () => {
  const tree = renderer.create(<Button name="me"></Button>).toJSON();
  expect(tree).toMatchSnapshot();
});
