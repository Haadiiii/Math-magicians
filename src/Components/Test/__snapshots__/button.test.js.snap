// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot for button 1`] = `
<div
  className="button"
>
  <button
    type="submit"
  >
    me
  </button>
</div>
`;
